moveit_core/trajectory_processing/test/test_linear_path_segment.cpp

```diff
@@ -0,0 +1,102 @@
+#include <gtest/gtest.h>
+#include <moveit/trajectory_processing/linear_path_segment.hpp>
+
+using trajectory_processing::LinearPathSegment;
+
+// Tests LinearPathSegment constructor.
+TEST(LinearPathSegment, Constructor)
+{
+  Eigen::VectorXd start(2);
```

Comment:
Can we test with fixed size vectors or are those not convertible to the `Xd` variable-sized vectors?